src/Compilers/CSharp/Test/Emit/Attributes/AttributeTests_Nullable.cs

```diff
@@ -463,14 +463,104 @@ static void G(A a, B b)
         // PROTOTYPE(NullableReferenceTypes): Execute some of these same tests with feature disabled.
 
         [Fact]
-        public void EmitAttribute_Constraint()
+        public void EmitAttribute_Constraint_Nullable()
+        {
+            var source =
+@"public class A
+{
+}
+public class B<T> where T : A?
+{
+}
+public class C<T> where T : A
+{
+}";
+            var comp = CreateCompilation(source, parseOptions: TestOptions.Regular8);
+            CompileAndVerify(comp, validator: assembly =>
+            {
+                var reader = assembly.GetMetadataReader();
+                var typeDef = GetTypeDefinitionByName(reader, "B`1");
+                var typeParameter = reader.GetGenericParameter(typeDef.GetGenericParameters()[0]);
+                var constraint = reader.GetGenericParameterConstraint(typeParameter.GetConstraints()[0]);
+                AssertAttributes(reader, constraint.GetCustomAttributes(), "MethodDefinition:Void System.Runtime.CompilerServices.NullableAttribute..ctor()");
+                typeDef = GetTypeDefinitionByName(reader, "C`1");
+                typeParameter = reader.GetGenericParameter(typeDef.GetGenericParameters()[0]);
+                constraint = reader.GetGenericParameterConstraint(typeParameter.GetConstraints()[0]);
+                AssertAttributes(reader, constraint.GetCustomAttributes());
+            });
+
+            var source2 =
+@"class Program
+{
+    static void Main()
+    {
+        new B<A?>();
```

Comment:
>B<A?> [](start = 12, length = 5)

I'd be tempted to also test permutations with a type derived from `A`, such as `new B<AChild?>();`, and so on.